doc/source/data_structures.rst

```diff
@@ -1,82 +1,516 @@
+
+
+Groups of atoms
+===============
+
+MDAnalysis has a hierarchy of :code:`Atom` containers that are used throughout the code.
+
+.. image:: images/classes.png
+
+First and foremost is the :code:`AtomGroup`. An :code:`AtomGroup` is the primary :code:`Atom` container; virtually everything can be accessed through it, as detailed `above <https://www.mdanalysis.org/UserGuide/data_structures.html#atomgroup>`_. This includes chemically meaningful groups of :code:`Atom`\ s such as a :code:`Residue` or a :code:`Segment`. 
+
+---------------------
+Residues and Segments
+---------------------
+
+A :code:`Residue` is composed of :code:`Atom`\ s, and a :code:`Segment` is composed of :code:`Residue`\ s.
+
+The corresponding container groups are :code:`ResidueGroup` and :code:`SegmentGroup`. These have similar properties and available methods as :code:`AtomGroup`.
+
+Each of these container groups can be accessed through another. For example:
+
+.. code-block::
+
+    >>> ag = u.atoms.select_atoms('resname ARG and name CA')
+    >>> ag
+    <AtomGroup with 13 atoms>
+    >>> ag.residues
+    <ResidueGroup with 13 residues>
+    >>> ag.residues.atoms
+    <AtomGroup with 312 atoms>
+    >>> ag.segments
+    <SegmentGroup with 1 segment>
+    >>> ag.segments.atoms
+    <AtomGroup with 3341 atoms>
+
+Similarly, an :code:`Atom` has direct knowledge of the :code:`Residue` and :code:`Segment` it belongs to.
+
+.. code-block::
+
+    >>> a = u.atoms[0]
+    >>> a.residue
+    <Residue MET, 1>
+    >>> a.residue.segment
+    <Segment 4AKE>
+    >>> a.residue.segment.residues
+    <ResidueGroup with 214 residues>
+
+-----------------------
+Fragments and molecules
+-----------------------
 
+Unlike Residues and Segments, fragments and molecules are not classes in MDAnalysis. A fragment is defined by bond connectivity. A fragment is what is typically considered a molecule: a group of atoms where each atom is bonded to at least one other atom in the fragment, and are not bonded to any atoms outside the fragment. The fragments of a Universe are determined by MDAnalysis as a derived quantity.
 
-ResidueGroup and SegmentGroup
-=============================
+The fragments of an :code:`AtomGroup` are accessible via the :attr:`fragments` property. In the case below, there is only one fragment in the Universe:
 
+.. code-block::
 
-For convenience, chemically meaningful groups of Atoms such as a 
-Residue or a Segment (typically a whole molecule or all of the solvent) 
-also exist as containers, as well as groups of these units (
-ResidueGroup, SegmentGroup).
+    >>> u.atoms.fragments
+    (<AtomGroup with 3341 atoms>,)
 
+In MDAnalysis, a molecule is a GROMACS-only concept. A group of atoms is considered a "molecule" if it is defined by the :code:`[ moleculetype ]` section in a GROMACS topology. Molecules are only defined if a Universe is created from a GROMACS topology. 
```

Comment:
@jbarnoud @richardjgowers how do we access molecules in the following example?
```python
import MDAnalysis as mda
from MDAnalysisTests import datafiles as data
u = mda.Universe(data.TPR, data.XTC)
```

I see `atoms.molnums` and `atoms.moltypes`. Are the "molecules" the "fragments"? 

Basically, how do use the "molecule" information?

Reply:
```python
>>> tpr = mda.Universe(TPR)
>>> tpr.atoms.molnums
array([    0,     0,     0, ..., 11086, 11087, 11088])
>>> tpr.atoms.moltypes
array(['AKeco', 'AKeco', 'AKeco', ..., 'NA+', 'NA+', 'NA+'], dtype=object)
```

This is my current understanding: like segments, molecules are read from the TPR file. `moltype` is the name of the molecule given immediately after the `[ moleculetype ]` header in the [human-readable topology files](http://manual.gromacs.org/archive/5.0/online/top.html). `molnum` is the order of the molecules in the `[ molecules ]` section. They're only available for universes created from tpr files; gro files don't include this information. Some methods can be applied to molecules. There's relevant discussion in #2 .

Fragments are MDAnalysis derived groups of atoms that are connected by bonds. They're available for all formats. Unsure what happens when you load in a file without connectivity information and then `guess_bonds` later -- do Fragments get re-computed?

A better name for `moltypes` might be `molnames` because GROMACS already has a molecule type concept with `residuetypes.dat`, which is a mapping of molecule names to Protein/DNA/RNA/water/ion categories.

Reply:
So the `molnum`/`moltypes` are just attributes and do not create any objects. Then it should definitely not show up in the figure.

> A better name for moltypes might be molnames because GROMACS already has a molecule type concept with residuetypes.dat, which is a mapping of molecule names to Protein/DNA/RNA/water/ion categories.

Yes, I agree, especially as "moleculetype" is the category. Maybe "moleculetype.name" ;-). (No)